### PR TITLE
[*] Smart Fields - Avoid crash from search

### PR DESCRIPTION
**Pull request by lucasscariot**

### CODE REVIEW

**Comment by lucasscariot** [comment]
[Lumberjacks want to search on activity logs](https://trello.com/c/WMo1bsZx/1438-lumberjacks-want-to-search-on-activity-logs)

File: services/resources-getter.js [hunk 1]
```diff
@@ -262,7 +262,11 @@ function ResourcesGetter(model, opts, params) {
     if (params.search) {
       _.each(schema.fields, function (field) {
         if (field.search) {
-          field.search(findAllOpts, params.search);
+          try {
+            field.search(findAllOpts, params.search);
+          } catch (error) {
+            console.log(error);
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: You need to use the logger util for that:
```
Interface.logger.error('Cannot search properly on Smart Field ' + field.field + ':\n' + error);
```
Something like this?

File: services/resources-getter.js [hunk 2]
```diff
@@ -283,7 +287,11 @@ function ResourcesGetter(model, opts, params) {
     if (params.search) {
       _.each(schema.fields, function (field) {
         if (field.search) {
-          field.search(countOpts, params.search);
+          try {
+            field.search(countOpts, params.search);
+          } catch (error) {
+            console.log(error);
+          }
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: You duplicate the code doing this (the current code is already duplicated...), I think it is time for code factorisation.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Fixed
+- Smart Field - Avoid crash from search and send the answer without failed smart field
```
Comment (arnaudbesnier) [inline CHANGELOG.md]:
```
### Added
- Smart Fields - Add an explicit error message if the search on a Smart Field generates an error.

### Fixed
- Smart Fields - A search on a collection having Smart Fields with search method implemented will respond properly (bypassing failing Smart Fields search if any).

**Pull request merged**